test/core/models/ocelot/stanzas/SequenceStanzaSpec.scala

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package core.models.ocelot.stanzas
+
+import base.BaseSpec
+import play.api.libs.json._
+import core.models.ocelot.Phrase
+
+class SequenceStanzaSpec extends BaseSpec {
+
+  val json: JsValue = Json.parse(
+    s"""|{
+        |    "next": ["1", "2", "3", "4", "end"],
+        |    "stack": false,
+        |    "options": [1,2,3,4],
+        |    "text": 0,
+        |    "label": "Items",
+        |    "type": "SequenceStanza"
+        |}""".stripMargin
+  )
+
+  val invalidJson: JsValue = Json.parse(
+    s"""|{
+        |    "type": "SequenceStanza",
+        |    "text": 0,
+        |    "next": ["2", "3", "4","end"],
+        |    "options": [1,2,3,4],
+        |    "label": "Items",
+        |    "stack": false
+        |}""".stripMargin
+  )
+
+  val expectedStanza: SequenceStanza =
+    SequenceStanza(0, Seq("1","2","3","4","end"), Seq(1,2,3,4), Some("Items"), stack = false)
+
+
+  "Reading valid JSON" should {
+    "create a Sequence Stanza" in {
+      json.as[SequenceStanza] shouldBe expectedStanza
+    }
+  }
+
+  "Reading invalid JSON" should {
+    "generate a JsError" in {
+      invalidJson.validate[SequenceStanza] match {
+        case JsError(_) => succeed
+        case _ => fail("An instance of SequenceStanza should not be created when the next and options list lengths differ")
+      }
+    }
+  }
+
+  "serialise to json" in {
+    Json.toJson(expectedStanza).toString shouldBe """{"text":0,"next":["1","2","3","4","end"],"options":[1,2,3,4],"label":"Items","stack":false}"""
+  }
+
+  "serialise to json from a Stanza reference" in {
+    val stanza: Stanza = expectedStanza
+    Json.toJson(stanza) shouldBe json
+  }
+
+  "SequenceStanza " should {
```

Comment:
Should this be "Sequence" rather than "SequenceStanza"?

Reply:
yes will change